src/libraries/System.Private.CoreLib/src/System/Number.NumberToFloatingPointBits.cs

```diff
@@ -419,57 +1164,78 @@ private static ushort NumberToHalfFloatingPointBits(ref NumberBuffer number, in
             uint integerDigitsPresent = Math.Min(positiveExponent, totalDigits);
             uint fractionalDigitsPresent = totalDigits - integerDigitsPresent;
 
-            uint fastExponent = (uint)(Math.Abs(number.Scale - integerDigitsPresent - fractionalDigitsPresent));
+            int exponent = (int)(number.Scale - integerDigitsPresent - fractionalDigitsPresent);
+            int fastExponent = Math.Abs(exponent);
 
-            // When the number of significant digits is less than or equal to 15 and the
-            // scale is less than or equal to 22, we can take some shortcuts and just rely
-            // on floating-point arithmetic to compute the correct result. This is
-            // because each floating-point precision values allows us to exactly represent
-            // different whole integers and certain powers of 10, depending on the underlying
-            // formats exact range. Additionally, IEEE operations dictate that the result is
-            // computed to the infinitely precise result and then rounded, which means that
-            // we can rely on it to produce the correct result when both inputs are exact.
+            // Above 19 digits, we rely on slow path
+            if (totalDigits <= 19)
+            {
+                byte* src = number.GetDigitsPointer();
 
-            byte* src = number.GetDigitsPointer();
+                if ((totalDigits <= 7) && (fastExponent <= 10))
```

Comment:
It might be worth profiling if this path is still worth having. It may be faster to remove the additional branch and just rely on the updated path instead.

Reply:
(Same for float/double)

Reply:
This is particularly given that the `if ((totalDigits <= 15) && (fastExponent <= 22))` path was removed.

If the new path is fast enough that doing the simple multiplication/division isn't worthwhile, then I expect this isn't as beneficial anymore.

Reply:
> It might be worth profiling if this path is still worth having

@tannergooding  I'm not sure I've got the proper setup to measure such particulary fine optimisation at home.

Reply:
If I read this code right, we are talking about Clinger's fast path. If I am correct, then @CarlVerret, it is easy enough to test it out: just general an input that falls on Clinger's fast path (entirely). Hopefully, we already have such a benchmark?

I think we can always surely answer the question: Clinger's fast path is absolutely still worth it.

The general rationale is like so...

1. The branch is likely predictable.
2. When it applies, you do get much faster processing (it becomes almost trivial in terms of instruction count).

Reply:
It is actually a remarkably elegant set of processing paths...

1. Clinger's fast path applies quite often... e.g., small integer values... and when it does, it is super fast.
2. When Clinger's fast path does not apply, we try our fast path which applies almost all of the time (99.9%) and is slower, but still quite fast.
3. If not, we fall back on something that is slower, but it is ok because it is uncommon in practice.

Reply:
(I understand that there might be an extra fast-path, if so, my default stance would be that it should be removed.)

Reply:
> (I understand that there might be an extra fast-path, if so, my default stance would be that it should be removed.)

Right, we actually have two versions of the fast path today (one for "can be done using `float`" and one for "can be done using `double`"), I expect that with the general `FastFloat` changes we can limit this to just "can be done using `double`".

Reply:
done.